pkg/image/controller/scheduler.go

```diff
@@ -69,16 +69,12 @@ func (s *Scheduler) next() (interface{}, interface{}, bool) {
 	defer s.mu.Unlock()
 
 	last := s.buckets[s.position]
-	if len(last) == 0 {
-		s.position = s.at(1)
-		last = s.buckets[s.position]
-	}
-
 	for k, v := range last {
 		delete(last, k)
 		s.buckets[s.at(-1)][k] = v
 		return k, v, false
 	}
+	s.position = s.at(1)
```

Comment:
in the interest of (some) clarity, i think i'd be inclined to move this into the "if len(last) == 0" block above and return from within that if-check.  The fact that the for-loop above is really an "if" statement makes it a bit non-obvious what the code flow is here.

Reply:
I tried it and don't believe it ends up clearer.  The compiler ends up insisting on a return statement at the end of the function which is actually unreachable.  Or you need to declare k and v outside the range and break.  I've added comments.